### PR TITLE
Backend Remediations in response to the Sprint 3 PR review

### PR DESCRIPTION
**Pull request by RafterGit**
### This pull request changes...

- adding more test cases to the `v1/users/set_profile` endpoint test suite
- made the logic around creating a user session in login.py DRYer
- expanded test coverage of `v1/logout/oidc` endpoint
- added a new first step for local development in respect to users who may experience file permission issues

### [AC] This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] Vulnerability Scan Test (Zap) has no high/critcial CVEs
- [ ] Linting Tests (ESLint for UI and Flake8 for Python)
- [ ] Unit Tests ( including a code coverage test report)
- [ ] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### [DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: tdrs-backend/tdpservice/users/test/test_api.py [hunk 1]
```diff
@@ -193,3 +193,43 @@ def test_set_profile_data_last_name_with_tilde(api_client, user):
...
+            },
+        )
+        assert response.status_code == status.HTTP_200_OK
+        """Test to ensure response data does not include unknown field"""
+        assert response.data == {"first_name": "Heather", "last_name": "Class"}
+        user.refresh_from_db()
+        assert user.first_name == "Heather"
+        assert user.last_name == "Class"
+        """Test fails if AttributeError exception isn't thrown"""
+        assert user.middle_name == "Unknown"
```
Comment (carltonsmith) [inline tdrs-backend/tdpservice/users/test/test_api.py:219]: Shouldn't there be an AttributeError when the endpoint receives a request with extra info, as opposed to when trying to access a field in the database that doesn't exist? Or am I reading this wrong?

File: tdrs-backend/tdpservice/users/test/test_api.py [hunk 2]
```diff
@@ -193,3 +193,43 @@ def test_set_profile_data_last_name_with_tilde(api_client, user):
...
+@pytest.mark.django_db
+def test_set_profile_data_missing_last_name_field(api_client, user):
+    """Test profile data cannot be set if last name field is missing."""
+    api_client.login(username=user.username, password="test_password")
+    response = api_client.post("/v1/users/set_profile/", {"first_name": "Heather", },)
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+def test_set_profile_data_missing_first_name_field(api_client, user):
```
Comment (carltonsmith) [inline tdrs-backend/tdpservice/users/test/test_api.py:231]: Is there anything to cover if the field is there, but the value is blank? like `{"first_name": "", "last_name": "Smith"}`

File: tdrs-backend/tdpservice/users/test/test_auth.py [hunk 3]
```diff
@@ -66,15 +67,20 @@ def test_oidc_auth(api_client):
     assert response.status_code == status.HTTP_302_FOUND
 
 
-def test_oidc_logout(api_client):
-    """Test logout url redirects."""
+def test_oidc_logout_without_token(api_client):
+    """Test logging out with token redirects and token is removed."""
```
Comment (carltonsmith) [inline tdrs-backend/tdpservice/users/test/test_auth.py:71]: the method name says "logout without token" but the doc string says "logging out with token and token is removed" but the test doesn't seem to mock a token or check for a removed token. Maybe I am missing something?

Comment (carltonsmith) [reply tdrs-backend/tdpservice/users/test/test_auth.py:71]: Ok - I think it's just the docstring that is wrong here as I see the logout with token below

**Review by carltonsmith — CHANGES_REQUESTED** [review]
Hey @RafterGit - Couple small changes I think, but that's all

File: tdrs-backend/tdpservice/users/test/test_api.py (hunk 1 above)
Comment (RafterGit) [reply tdrs-backend/tdpservice/users/test/test_api.py:219]: @carltonsmith  by default django will accept what ever you throw at it , but if it isn't part of the model it gets discarded when saving to it / updating it

File: tdrs-backend/tdpservice/users/test/test_api.py (hunk 2 above)
Comment (RafterGit) [reply tdrs-backend/tdpservice/users/test/test_api.py:231]: This block of code should handle that 

https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/tdrs-backend/tdpservice/users/test/test_api.py#L84-L111

File: tdrs-backend/tdpservice/users/test/test_auth.py (hunk 3 above)
Comment (RafterGit) [reply tdrs-backend/tdpservice/users/test/test_auth.py:71]: Ill update the docstring to be less confusing

File: tdrs-backend/tdpservice/users/test/test_api.py (hunk 1 above)
Comment (carltonsmith) [reply tdrs-backend/tdpservice/users/test/test_api.py:219]: 👍

**Review by carltonsmith — APPROVED** [review]
Looks good to me

**Comment by RafterGit** [comment]
> This looks good, should I also wait to look at #262 merged into this or review separately?

@lauraGgit  since those are independent of the backend changes please review it separately.

**Review by lauraGgit — APPROVED** [review]

**Pull request merged**